src/core/Akka.Tests/Actor/ActorTelemetrySpecs.cs

```diff
@@ -0,0 +1,304 @@
+                Assert.Equal(102, telemetry.ActorCreated);
+                Assert.Equal(100, telemetry.ActorRestarted);
+                Assert.Equal(101, telemetry.ActorStopped);
+            }, RemainingOrDefault);
+        }
+        
+        // create a unit test where a parent actor spawns 100 children and then restarts
+        [Fact]
+        public async Task ActorTelemetry_must_be_accurate_when_parent_restarts()
+        {
+            // create a TelemetrySubscriber actor
+            var subscriber = Sys.ActorOf(Props.Create<TelemetrySubscriber>(), "subscriber");
+            
+            // request current telemetry values (ensure that the actor has started, so counter values will be accurate)
+            await subscriber.Ask<TelemetrySubscriber.GetTelemetry>(TelemetrySubscriber.GetTelemetryRequest.Instance);
+            
+            // create a parent actor
+            var parent = Sys.ActorOf(Props.Create<ParentActor>(), "parent");
+
+            // send a message to the parent to create 100 children
+            parent.Tell(new CreateChildren(100));
+
+            // wait for the parent to reply back
+            ExpectMsg("done");
+            
+            // awaitassert collecting data from the telemetry subscriber until we can see that 102 actors have been created
+            // 100 children + parent + subscriber itself
+            await AwaitAssertAsync(async () =>
+            {
+                var telemetry = await subscriber.Ask<TelemetrySubscriber.GetTelemetry>(TelemetrySubscriber.GetTelemetryRequest.Instance);
+                Assert.Equal(102, telemetry.ActorCreated);
+                // assert no restarts or stops recorded
+                Assert.Equal(0, telemetry.ActorRestarted);
+                Assert.Equal(0, telemetry.ActorStopped);
+            }, RemainingOrDefault);
+            
+            // send a message to the parent to restart
+            parent.Tell("restart");
+            
+            // await assert collecting data from the telemetry subscriber until we can see that 102 actors have been restarted
+            await AwaitAssertAsync(async () =>
+            {
+                var telemetry = await subscriber.Ask<TelemetrySubscriber.GetTelemetry>(TelemetrySubscriber.GetTelemetryRequest.Instance);
+                // assert that actor start count is still 102
+                Assert.Equal(102, telemetry.ActorCreated);
+                
+                // only 1 parent restart recorded
+                Assert.Equal(1, telemetry.ActorRestarted);
+                // assert 100 stops recorded (only the child actors)
+                Assert.Equal(100, telemetry.ActorStopped);
+            }, RemainingOrDefault);
+        }
+        
+        /// <summary>
+        /// Pool routers should have their start / stop / restarts counted too
+        /// </summary>
+        [Fact]
+        public async Task ActorTelemetry_must_be_accurate_for_pool_router()
```

Comment:
Router actors have an `ActorCell` even though they don't have a typical `Mailbox`, therefore they will also emit the same events are other local actors do for starting / stopping / restarting.